internal/server/data/migrations_test.go

```diff
@@ -904,6 +904,12 @@ func parseTime(t *testing.T, s string) time.Time {
 	return v
 }
 
+func runStep(t *testing.T, name string, fn func(t *testing.T)) {
+	if !t.Run(name, fn) {
```

Comment:
what does this do?

Reply:
I moved this from another file because it's now only used in `migrations_test.go`. That's why it's showing up in the diff. It was previously in another file.

This runs a function similar to `t.Run`, however if that test function fails it stops the parent test case. Normally a failure in `t.Run` will continue to run the rest of the parent test case. 

This is useful for multi-step tests, where you want some of the benefits of `t.Run` (using `t.Cleanup` to cleanup when the subtest exits, and the additional context of the `name` in the failure message), without the behaviour of continuing to run the remaining subtests.